web/src/test/webapp/spec/routeManipulationSpec.js

```diff
@@ -0,0 +1,74 @@
+// because leaflet-src.js requires window, document etc. we need to provide these variables...
```

Comment:
This is pretty ugly. Maybe you have a better idea ? We could keep the tests separate from leaflet entirely by reimplementing/copy&pasting the functionality of `L.latLng`, but I do not think this is much nicer. We could also put this workaround into a separate file.